### ISSUE TITLE
Errors in tf.Session().run or tf.reduce_max

### ISSUE THREAD

**Issue by Zlz-shoulder** [tensorflow/tensorflow#41445]
Question:
opt = sess.run(pred_DFS, feed_dict = {x: clinic_factors_val, keep_prob: 1.0, treatment: treat)}) 
opt1, Pat_pred = sess.run([output1,pred_DFS], feed_dict = {x: clinic_factors_val), keep_prob: 1.0, treatment: treat)}) 
print(opt1)
print(opt)
print(Pat_pred)
The relationship between pred_DFS and output1: pred_DFS = tf.reduce_max(output1,axis=1). output1 is an intermediate variable.

When pred_DFS is run alone, the result is wrong, but when pred_DFS and output1 are run together, the result of pred_DFS is correct. Why?

the result of a run：
opt1：[[0.23930925 0.36091098 0.61886203]]
opt：[0.72315866]
Pat_pred：[0.61886203]
Note that，the results are consistent when output1 is run alone and when pred_DFS and output1 are run together

**Comment by Saduf2019** [comment]
@Zlz-shoulder 
Please share complete stand alone code for us to replicate the issue faced along with tf version, or a colab gist with the issue faced for analysis.

**Comment by Zlz-shoulder** [comment]
> @Zlz-shoulder
> Please share complete stand alone code for us to replicate the issue faced along with tf version, or a colab gist with the issue faced for analysis.
I have shared complete stand alone code and the corrsponding data in https://github.com/Zlz-shoulder/test_errors/tree/master
In order to run ‘PH+BMC_matched_test.py’, you only need to assign the path of 'PH+BMC_matched_test.csv' to 'clinic_path' in the  ‘PH+BMC_matched_test.py’

**Comment by Zlz-shoulder** [comment]
I have run alone pred_DFS or output1 twice, the results maintain unchange. but when pred_DFS and output1 are run together, the value of output1 is the same as the previous operation, but it is not true for pred_DFS.  It is so weird！

**Comment by Zlz-shoulder** [comment]
I changed this function and it's right now: pred_DFS = tf.py_func(np.max, [output1,1], tf.float32).  Is ‘tf.reduce_max’ something wrong?！

**Comment by Saduf2019** [comment]
@Zlz-shoulder 
I ran the code shared and [face a different error](https://colab.research.google.com/gist/Saduf2019/7d000818b3f287e7fc63e73155b6c894/untitled284.ipynb), please share all dependencies or share a colab gist for us to analyse the issue faced.

**Comment by Zlz-shoulder** [comment]
You must install lifelines! All dependencies are as follows:
pandas                    1.0.3            py36h0573a6f_0
tensorboard               1.12.2           py36he6710b0_0 
tensorflow                1.12.0          gpu_py36he68c306_0
tensorflow-base           1.12.0          gpu_py36h8e0ae2d_0 
tensorflow-gpu            1.12.0               h0d30ee6_0
numpy                     1.18.1           py36h4f9e942_0
numpy-base                1.18.1           py36hde5b4d6_1
lifelines                 0.24.6                    <pip>
scipy                     1.4.1            py36h0b6359f_0

**Comment by Saduf2019** [comment]
@Zlz-shoulder 
I ran the code shared with the dependencies shared, please [find the error faced.](https://colab.research.google.com/gist/Saduf2019/6ff7fed10a67b4776e3e9267704e26a3/untitled283.ipynb), please share a colab gist with the issue reported.

**Comment by Zlz-shoulder** [comment]
I fail to log in my Google account, and I also fail to create a new Google account. But I believe this is a simple problem, and you could find many similar errors on the Internet. In addition. I use the Tensorflor of 1.12.0.

**Comment by Saduf2019** [comment]
@Zlz-shoulder 
I have resolved that error, and a different error occurred , please refer to this [gist here](https://colab.research.google.com/gist/Saduf2019/0977600b4f838c33b711e738da4d238c/untitled283.ipynb).

**Comment by Zlz-shoulder** [comment]
Please delete the line 339-340, which are not important. Many thanks.

**Comment by Saduf2019** [comment]
@Zlz-shoulder 
Can you please mention the lines here, so its easier or delete them and share the gist.

**Comment by Zlz-shoulder** [comment]
Please delete the lines 339-340: 
with open(log_path, 'a') as f:
         f.write(line + '\n')

**Comment by Saduf2019** [comment]
@Zlz-shoulder 
PLease let us know if this confirms your issue, as per [gist here.](https://colab.research.google.com/gist/Saduf2019/995fc0057b705d6880bc4da600a207e6/untitled283.ipynb)

**Comment by Zlz-shoulder** [comment]
Yes, but your results do not have the questions that I encountered. This may be because you use TensorFlow of version 1.1.15 while I use TensorFlow of version 1.1.12.  In my settings, the output values of pred_DFS are different between When pred_DFS is run alone and when pred_DFS and output1 are run together. Why?

**Comment by Zlz-shoulder** [comment]
Is there no one responding here?

**Comment by gowthamkpr** [comment]
@Zlz-shoulder Sorry for the late response. Can you please use the latest version on tensorflow 1.x i.e., tensorflow 1.15 and see if the issue still persists? Thanks!

**Comment by Zlz-shoulder** [comment]
You had tested it on tensorflow 1.1.15 and found no errors, so I think that I do not need to test it on tensorflow 1.1.15. However, I found errors using the exactly the same code on tensorflow 1.1.12. In some of the first comments, I have described in detail the errors I encountered about tensorflow 1.1.12. Could you help me answer my question？

**Comment by gowthamkpr** [comment]
@Zlz-shoulder This is likely been fixed in 1.15. So, that is the reason why we recommend using tensorflow 1.14 or later. Thanks!

**Comment by Zlz-shoulder** [comment]
Many thanks! But I hope that you can help me comfirm whether it is an error in tensorflow 1.12  or an error with my codes. Thanks!

**Comment by gowthamkpr** [comment]
@Zlz-shoulder This error is with tensorflow 1.12 and is been fixed in future versions. I would recommend you to use tensorflow versions > 1.14. Thanks!

**Comment by Zlz-shoulder** [comment]
OK. Many thanks!

**Comment by gowthamkpr** [comment]
Closing this issue as it has been resloved. Please add additional comments for us to open this issue again. Thanks!
